### PR TITLE
[bug] Fix User profile update

### PR DESCRIPTION
**Pull request by theBeginner86**
**Notes for Reviewers**

Possible reasons:

<del>1. Request body wasn't passed in a few scenarios</del>
2. Appropriate Headers weren't added to the new request so the identity server rejected the request to update the profile

Solution:
This PR refactored the code a bit to take care of these scenarios.

This PR fixes # https://layer5io.slack.com/archives/C02F90W866T/p1671729067942749?thread_ts=1671482002.636389&cid=C02F90W866T

https://user-images.githubusercontent.com/73700530/209432141-8f68cacc-17bf-4424-b89c-d7b41a04779a.mov

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: server/models/remote_provider.go [hunk 1]
```diff
@@ -3024,34 +3024,32 @@ func (l *RemoteProvider) ExtensionProxy(req *http.Request) ([]byte, error) {
...
+	
+	defer req.Body.Close()
+	bd, _ := io.ReadAll(req.Body)
+	bf := bytes.NewBuffer(bd)
 
 	// make http.Request type variable with the constructed URL
-	cReq, _ := http.NewRequest(req.Method, remoteProviderURL.String(), req.Body)
-	tokenString, err := l.GetToken(req)
-
+	cReq, err := http.NewRequest(req.Method, remoteProviderURL.String(), bf)
```
Comment (MUzairS15) [inline server/models/remote_provider.go:3033]: Do you know why body was not passed correctly?
I think the issue was only bcz of incorrect headers.

**Review by MUzairS15 — APPROVED** [review]

File: server/models/remote_provider.go (hunk 1 above)
Comment (theBeginner86) [reply server/models/remote_provider.go:3033]: Hmm, I'm not sure. While debugging I realised that an empty body was being sent in the request to Meshery Cloud. So on other side there were sometimes error related to unmarshalling as the body itself was empty
So to take care of any edge case, I kind of added this extra step

_But we have been using these user extensions for a while and haven't seen that error. 🤔_

Comment (MUzairS15) [reply server/models/remote_provider.go:3033]: After fixing headers, did you tried without reading and sending body explicitly?

Comment (theBeginner86) [reply server/models/remote_provider.go]: Yeah, now it seems to work fine without the conversion of the original body. I might have bad build while debugging.
So I reverted that change.

**Comment by leecalcote** [comment]
haha! What a nice debug message. 🎄

**Comment by leecalcote** [comment]
@MUzairS15 round two of feedback / review needed.

**Review by MUzairS15 — APPROVED** [review]

**Pull request merged**